frontend/src/utils/utils.js

```diff
@@ -0,0 +1,30 @@
+import { useRef, useEffect } from "react";
+/**
+ * Tracks the previous value of the given item. Returns
+ * the previous value
+ * @param value : The object to be tracked
+ */
+
+export function usePrevious(value) {
+  const ref = useRef();
+  useEffect(() => {
+    ref.current = value;
+  });
+  return ref.current;
+}
+
+/**
+ * Replaces the spaces in a given string with hyphens.
+ * @param {string} urlString The string to be converted from spaces to hyphens.
+ */
+export function replaceSpaceCharacters(urlString) {
```

Comment:
I think we can just use Python's own replace() and strip() methods. Won't be more expensive and will look cleaner

Reply:
Done